File: swift/hanoi_tower.swift
func hanoi(_ n: Int, _ from: String, _ mid: String, _ to: String) -> Int {
  n == 1 ? 1 : hanoi(n - 1, from, to, mid) + 1 + hanoi(n - 1, mid, from, to)
}

print("\(hanoi(32, "a", "b", "c")) moves")

File: rust/hanoi_tower.rs
fn hanoi(n: u32, from: &char, mid: &char, to: &char) -> u32 {
  if n == 1 {
    return 1;
  }
  return hanoi(n-1, from, to, mid) + 1 + hanoi(n-1, mid, from, to);
}

fn main() {
    let moves = hanoi(32, &'a', &'b', &'c');
    println!("{} moves", moves);
}
